### DISCUSSION TITLE
Not able to access NEXT_PUBLIC_APIKEY when configured in kubernates

### DISCUSSION THREAD

**Discussion by suri66** [vercel/next.js#14812] [Help]
As per version 9.4, environment variables are available to both server and browser if we configured those by prefixing NEXT_PUBLIC. So this is my env file.

**.env**

```
NEXT_PUBLIC_APIKEY=<value>
NEXT_PUBLIC_GRAPHQL_URL=<value>
NEXT_PUBLIC_INDABODE_WEBSITE_URL=<value>
```

This is working fine in local. I have tested using the build as well.

On the server, I have configured these var in the Kubernetes itself. In this case env var not available to my app.

Previously I have an old version of Next.js. There also I have set the env var in the Kubernates, but those were available to the app. I have mentioned the details below.

**.env**

```
APIKEY=<value>
GRAPHQL_URL=<value>
INDABODE_WEBSITE_URL=<value>
```

**next.config.js**

```
const webpack = require('webpack');

module.exports = {
  webpack(config) {
  
    config.plugins.push(new webpack.DefinePlugin({
      'process.env.APIKEY': JSON.stringify(process.env.APIKEY),
      'process.env.GRAPHQL_URL': JSON.stringify(process.env.GRAPHQL_URL),
      'process.env.INDABODE_WEBSITE_URL': JSON.stringify(process.env.INDABODE_WEBSITE_URL),
    }));
        
    return config;
  }
};
```

I don't know what is the problem with the latest version 9.4 I have not set up anything in next.config.js for old version not for new. As per version 9.4 doc, env var available to both server and browser if we use NEXT_PREFIX.

Am I missing something?

**Comment by Fumler** [comment]
You need to have the environment variables accessible wherever you are running `next build` I presume. I assume you are not running `next build` inside kubernetes.

**Reply by suri66** [reply]
Hi @Fumler,  `next build` command is there. The application is working fine. I can see the web page working. But things which depend on env are not working.

**Reply by Fumler** [reply]
In the environment where you are running your `next build` command, is e.g. `NEXT_PUBLIC_APIKEY` set in that environment?

**Reply by suri66** [reply]
I have used Dockerfile so all things are running inside a container.. using kubernates also

This is kubernates yaml  where I have set env var, no .env file on server. All env var are set here
        
```
         env:

        - name: NEXT_PUBLIC_APIKEY

          value: <value>

        - name: NEXT_PUBLIC_GRAPHQL_URL

          value: <value>

        - name: NEXT_PUBLIC_INDABODE_WEBSITE_URL

          value: <value>

```

The same kubernates yaml was used previously and the app was able to access env var.  But not now in the latest version where we have NEXT_PUBLIC feature.
